DEVELOPMENT.md

```diff
@@ -0,0 +1,229 @@
+# Setting up a Development Environment
+
+This document details how to set up a local development environment that will allow you to contribute changes to the project!
+
+## Base Requirements
+
+* The project is hosted on GitHub, so you need an account there (and if you are reading this, you likely do!)
+* An IDE such as Microsoft VS Code IDE https://code.visualstudio.com/
+
+## Set up Git Repository Fork
+
+You will push changes to a fork of the Langflow repository, and from there create a Pull Request into the project repository.
+
+Fork the [Langflow GitHub repository](https://github.com/langflow-ai/langflow/fork), and follow the instructions to create a new fork.
+
+On your new fork, click the "<> Code" button to get a URL to clone using your preferred method, and clone the repostory; for example using `https`:
+
+```bash
+git clone https://github.com/<your username>/langflow.git
+```
+
+Finally, add the Project repository as `upstream`:
+
+```bash
+cd langflow
+git remote add upstream https://github.com/langflow-ai/langflow.git
+git remote set-url --push upstream no_push
+```
+
+> **Windows/WSL Users**: You may find that files "change", specifically the file mode e.g. "changed file mode 100755 → 100644". You can workaround this problem with `git config core.filemode false`.
+
+## Set up Environment
+
+There are two options available to you: the 'easy' and recommended option is to use a Development Container ("[Dev Container](https://containers.dev/)"), or you can choose to use your own OS / environment.
+
+### Option 1 (Preferred): Use a Dev Container
+
+Open this repostiory as a Dev Container per your IDEs instructions.
+
+#### Microsoft VS Code
+
+* See [Developing inside a Container](https://code.visualstudio.com/docs/devcontainers/containers)
+* You may also find it helpful to [share `git` credentials](https://code.visualstudio.com/remote/advancedcontainers/sharing-git-credentials) with the container
+
+
+### Option 2: Use Your Own Environment
+
+Install Pre-Requisites:
+
+* **Operating System**: macOS or Linux; Windows users ***MUST*** develop under WSL.
+* **`git`**: The project uses the ubiquitous `git` tool for change control.
+* **`make`**: The project uses `make` to coordidinate packaging.
+* **`uv`**: This project uses `uv` (`>=0.4`), a Python package and project manager from Astral. Install instructions at https://docs.astral.sh/uv/getting-started/installation/.
+* **`npm`**: The frontend files are built with Node.js (`v22.12 LTS`) and `npm` (`v10.9`). Install instrucations at https://nodejs.org/en/download/package-manager.
+  * Windows (WSL) users: ensure `npm` is installed within WSL environment; `which npm` should resolve to a Linux location, not a Windows location.
```

Comment:
Maybe there's an extra space here?
```suggestion
 * Windows (WSL) users: ensure `npm` is installed within WSL environment; `which npm` should resolve to a Linux location, not a Windows location.
```

Reply:
I've switched this to a `-`, it is meant to be an indented sublist as it applies to `npm` but only Windows users.